### ISSUE TITLE
Add disk usage information

### ISSUE THREAD

**Issue by dbwiddis** [dblock/oshi#38]
Cross platform, Java 1.6:
```java
    File[] roots = File.listRoots();
    for (File root : roots) {
      double usedSpace = (double)(root.getTotalSpace() - root.getUsableSpace())/root.getTotalSpace();
      System.out.format("%s: %.1f used of %s%n", 
          root.getAbsolutePath(), usedSpace, FormatUtil.formatBytes(root.getTotalSpace()));
    }
```

**Comment by dbwiddis** [comment]
See http://stackoverflow.com/questions/462110/acquiring-drive-names-as-opposed-to-drive-letters-in-java for more useful drive names and descriptions (probably use root.toString() as a fallback for an empty name).  This works well for Windows, but only gives "/" in Mac/Linux so will need platform-specific implementations.

**Comment by dbwiddis** [comment]
Mac solution is easy: just list /Volumes.
Unix possibly search for /dev/disk*

**Comment by dbwiddis** [comment]
I've got a working cross-platform (see note) implementation with a FileSystem class that can be extended by the platform-specific versions. Windows will use the default implementation, Mac/Linux will hide the `getFileSystems()` method to parse /Voluumes and /proc/mounts, respectively.

Note: Mac/Linux work now with the "/" directory but should be able to improve on this:
```
File System:
 / (file system) 121.9 GB of 697.5 GB free (17.5%)
```
And I need a bigger hard drive, or to do some cleanup maintenance :grin:

**Comment by dbwiddis** [comment]
Mac is looking good... 
```
File System:
 Data (/Volumes/Data) 19.2 GB of 1.8 TiB free (1.0%)
 MacData (/Volumes/MacData) 3.4 GB of 4.4 GB free (77.4%)
 Macintosh HD (/Volumes/Macintosh HD) 122.1 GB of 697.5 GB free (17.5%)
 MobileBackups (/Volumes/MobileBackups) 0 bytes of 697.5 GB free (0.0%)
 Time Machine Backups (/Volumes/Time Machine Backups) 19.1 GB of 1.8 TiB free (1.0%)
```

**Comment by dbwiddis** [comment]
Windows...
```

File System:
 Floppy Disk Drive (A:) (Floppy Disk Drive) 1.1 MB of 1.4 MB free (82.4%)
 Local Disk (C:) (Local Disk) 27.3 GB of 64.0 GB free (42.7%)
 D:\ (CD Drive) 0 bytes of 0 bytes free (NaN%)
 MobileBackups on 'psf' (T:) (Network Drive) 0 bytes of 697.5 GB free (0.0%)
 MacData on 'psf' (U:) (Network Drive) 3.4 GB of 4.4 GB free (77.4%)
 Photo Stream on 'psf' (W:) (Network Drive) 121.7 GB of 697.5 GB free (17.4%)
 iCloud on 'psf' (X:) (Network Drive) 121.7 GB of 697.5 GB free (17.4%)
 Home on 'psf' (Y:) (Network Drive) 121.7 GB of 697.5 GB free (17.4%)
 Host on 'psf' (Z:) (Network Drive) 121.7 GB of 697.5 GB free (17.4%)
```

**Comment by dbwiddis** [comment]
Linux... 
```
File System:
 / (/) 52.8 GB of 60.9 GB free (86.7%)
 Home (/media/psf/Home) 122.9 GB of 697.5 GB free (17.6%)
 MacData (/media/psf/MacData) 3.4 GB of 4.4 GB free (77.4%)
 MobileBackups (/media/psf/MobileBackups) 0 bytes of 697.5 GB free (0.0%)
```

**Comment by dbwiddis** [comment]
Merged in #50
